src/sentry/search/events/builder/metrics.py

```diff
@@ -376,14 +377,51 @@ def validate_aggregate_arguments(self) -> None:
         if not self.use_metrics_layer:
             super().validate_aggregate_arguments()
 
+    # This property is used to determine if a query is using at least one of the fields in the spans namespace.
+    @property
+    def is_spans_metrics_query(self) -> bool:
+        if self.query is not None:
+            tags = parse_query(
+                self.params.projects, self.query, self.params.user, self.params.environments
+            )["tags"]
+            for tag in tags:
+                if tag in constants.SPANS_METRICS_TAGS:
+                    return True
+        for column in self.selected_columns:
+            # Not using parse_function since it checks against function_converter
+            # which is not loaded yet and we also do not need it
+            match = fields.is_function(column)
+            func = match.group("function") if match else None
+            if func in constants.SPANS_METRICS_FUNCTIONS:
+                return True
+            argument = match.group("columns") if match else None
+            if argument in constants.SPAN_METRICS_MAP.keys() - constants.METRICS_MAP.keys():
+                return True
+        return False
+
+    # Some fields and functions cannot be translated to metrics layer queries.
+    # This property is used to determine if a query is using at least one of these fields or functions, and if so, we must not use the metrics layer.
+    @property
+    def is_unsupported_metrics_layer_query(self) -> bool:
```

Comment:
Could you cache this property? I'm pretty sure `use_metrics_layer` is called all over the place and we don't want to recalculate this determination repeatedly

Reply:
👍 added!